db/migrate/20220909105939_add_category_ref_to_products.rb

```diff
@@ -0,0 +1,6 @@
+class AddCategoryRefToProducts < ActiveRecord::Migration[7.0]
+  def change
+    add_column :products, :category_id, :integer, null: false
```

Comment:
can we use add_reference instead? How is it different?

Reply:
Yes, it can be used
add_column won't add foreign key as referential integrity, to do so we would need to write it explicity as done in the following line
add_reference has an optional which lets us add foreign key referential integrity in the database.

Reply:
changed it to add_reference